handler/src/main/java/io/netty/handler/timeout/IdleStateHandler.java

```diff
@@ -297,27 +318,38 @@ private void initialize(ChannelHandlerContext ctx) {
         }
 
         state = 1;
+        initOutputChanged(ctx);
 
-        EventExecutor loop = ctx.executor();
-
-        lastReadTime = lastWriteTime = System.nanoTime();
+        lastReadTime = lastWriteTime = tick(TimeUnit.NANOSECONDS);
         if (readerIdleTimeNanos > 0) {
-            readerIdleTimeout = loop.schedule(
-                    new ReaderIdleTimeoutTask(ctx),
+            readerIdleTimeout = schedule(ctx, new ReaderIdleTimeoutTask(ctx),
                     readerIdleTimeNanos, TimeUnit.NANOSECONDS);
         }
         if (writerIdleTimeNanos > 0) {
-            writerIdleTimeout = loop.schedule(
-                    new WriterIdleTimeoutTask(ctx),
+            writerIdleTimeout = schedule(ctx, new WriterIdleTimeoutTask(ctx),
                     writerIdleTimeNanos, TimeUnit.NANOSECONDS);
         }
         if (allIdleTimeNanos > 0) {
-            allIdleTimeout = loop.schedule(
-                    new AllIdleTimeoutTask(ctx),
+            allIdleTimeout = schedule(ctx, new AllIdleTimeoutTask(ctx),
                     allIdleTimeNanos, TimeUnit.NANOSECONDS);
         }
     }
 
+    /**
+     * This method is visible for testing! It's purpose is to decouple us from
+     * a clock that we can't control.
+     */
+    long tick(TimeUnit unit) {
```

Comment:
Can be done but is there a downside with the current impl? `TimeUnit#convert(...)` will be a ~noop if source and destination units are the same (which is the case inside the IdleStateHandler class).